crates/solidity/testing/snapshots/bindings_assertions/control/do_while.sol

```diff
@@ -0,0 +1,17 @@
+contract Test {
+    function test() public returns (int) {
+        int i = 1;
+        //  ^def:1
+        do {
+            int odd = i % 2;
+            //  ^def:2
+            //        ^ref:1
+            i *= 3;
+            //<ref:1
```

Comment:
Is this a typo? or is it intended?

```suggestion
            //^ref:1
```

Reply:
On a parallel, for these tests, since we don't have any other use case for comments, should we error out if a comment line doesn't have a specific assertion?
It might catch other typos like this.

Reply:
I found a few others like this. For example:
https://github.com/manastech/slang/blob/d86eefd2290c1e63ed0b75416f696507d1618a98/crates/solidity/testing/snapshots/bindings_assertions/control/for_empty_init_or_cond.sol#L11

Reply:
These are not typos, but a special syntax to indicate the column of the assertion is at the start of the comment. Otherwise, you would be asserting on the `*=` operator in this case.

Reply:
> On a parallel, for these tests, since we don't have any other use case for comments, should we error out if a comment line doesn't have a specific assertion? It might catch other typos like this.

I like this idea, but only if we add syntax to skip an assertion. I've been using `#ref:5` instead of the caret prefix for that, in cases where I want to mark assertions that I know I have to fix at some point (kinda like a `TODO` or `FIXME`). Maybe I can use the version requirements field and do something like `// ^ref:1 (skipped)`.

That way we can also keep track of skipped assertions more easily.

Reply:
I see. Thanks for explaining.

> Otherwise, you would be asserting on the *= operator in this case.

Is this because you have auto-formatting enabled? Can't we just shift the comment left? that would make it more consistent with the rest.
I think this was problematic for parser tests, so we disabled formatting for `.sol` files there, because some of them are incomplete/have errors by design.

> in cases where I want to mark assertions that I know I have to fix at some point (kinda like a TODO or FIXME).

If this is temporary (just as we are still building the bindings), maybe we can also use `// ^ref:1 (=0.0.0)` which won't match anything, and doesn't require special casing in the test runner.
Just a suggestion. Up to you if you prefer a different way.

Reply:
> Is this because you have auto-formatting enabled? Can't we just shift the comment left? that would make it more consistent with the rest.

More like auto indentation from my editor. It increases friction when writing tests, so I'd rather support the special syntax. With it, there is only a single column that cannot be addressed (the second column after the comment), but it's extremely rare to need that, and that can be worked around without messing up the natural indentation of lines.

> If this is temporary (just as we are still building the bindings), maybe we can also use // ^ref:1 (=0.0.0) which won't match anything, and doesn't require special casing in the test runner.
> Just a suggestion. Up to you if you prefer a different way.

Yeah, that works! I was looking for something more unique, like `ignore` or `skip` but those are common words. Searching for `=0.0.0` would actually provide an accurate list of "pending" assertions. Thanks!

I added a check for comments without assertions in the last commit pushed. There were a [couple of instances](https://github.com/NomicFoundation/slang/pull/1076/files#diff-dece60363a9e31f71477916a2d9a8b985447be2cd4a756768c16aa32f01d9ec2) where I was using comments but I inlined them with the assertion (or removed them because they were redundant).

Reply:
> It increases friction when writing tests, so I'd rather support the special syntax.

I see. How about using a slightly longer variable name then, so that it can be caught using the same `^` syntax?
Just trying to suggest a way to match this to all the existing tests/snapshots we use for bindings.

Not a blocker for this PR of course.

Reply:
It wouldn't catch it anyway, because the `^` requires matching the beginning of the identifier. We may want to relax that restriction and match anywhere inside the identifier. I wonder if that extra wiggle room for assertion declaration would let some subtle accidental mistake through... I'll play with that idea a bit. Thanks for the suggestion!